nf_core/download.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+""" Download a nf-core pipeline """
+
+from __future__ import print_function
+
+from io import BytesIO
+import logging
+import os
+import requests
+import subprocess
+import sys
+from zipfile import ZipFile
+
+
+import nf_core.list, nf_core.lint
+
+def download_workflow(pipeline, release=None, singularity=False, outdir=None):
+    """ Main function to download a nf-core workflow """
+
+    # init
+    wf = DownloadWorkflow(pipeline, release, singularity, outdir)
+
+    # Get workflow details
+    if not wf.get_workflow():
+        sys.exit(1)
+
+    # Check that the outdir doesn't already exist
+    if os.path.exists(wf.outdir):
+        logging.error("Output directory '{}' already exists".format(wf.outdir))
+        sys.exit(1)
```

Comment:
Do you really wanna exit ? If so why not have this check at the beginning (like as soon as receiving the parameter or in the class) that will save us all the trouble of doing everything 😇 

Or just warn and save in a custom name so they can rename the folder later ?

Reply:
Yeah, should probably handle this in a nicer way 👍

Reply:
Ok, I remember..

> If so why not have this check at the beginning

I can't check any earlier, as if `--outdir` is not specified, the default output directory name is generated based on the workflow name and version number. The version number is the last release if not specified. So basically, I don't know what the output directory name will be before I get the workflow details.

> Or just warn and save in a custom name so they can rename the folder later?

Nah - this command isn't going to be super common to run (like `multiqc` or `nextflow` or something), so it's not such a big deal to just bail and exit I think. Less code and potentially less confusing.

Reply:
Ok, I see the point now 👍